exercises/mongo-shell/basic-mongo-queries.md

```diff
@@ -514,33 +661,100 @@ db.movieDetails.find({"genres":["Action", "Adventure"]},
+    {<your uniquely identifying filter as key: value pair>}, 
+    {"$set": {"title": "Taare Zameen Par"}}
+    )
+```
 
-Explore the different datasets available on your workstation.  Be creative and try to find interesting things.  
+**Exercise 16** :computer: 
 
-Some questions to consider while exploring:
-- How is this data structured?
-- Is there anything interesing to me about this data?
-- Is the data malformed, incorrect or incomplete?
-- What are the challenges you faced while exploring this data?
+MongoDB offers multiple operators that can be used in its update operations. In the exercise above, we looked at the `$set` operator. 
 
-**Presentation: :alarm_clock: 25 minutes**
-Each group will prepare a small, **5 minute**, 2-3 slide presentation about the datasets.
+|Name|Description|
+|---|---|
+|`$set`|Sets the value of a field in a document|
+|`$unset`|Removes the specified field from a document|
+|`$currentDate`|Sets the value of a field to current date, either as a Date or a Timestamp|
+|`$inc`|Increments the value of a field by a specified amount|
+|`$mul`|Multiplies the value of the field by the specified amount|
+|`$min`|Only updates the field if the specified value is less than the existing field value.|
+|`$max`|Only updates the field if the specified value is greater than the existing field value.|
+|`$setOnInsert`|Sets the value of a field if an update results in an insert of a document. Has no effect on update operations that modify existing documents.|
+|`$rename`|Renames a field|
+
+
+Take **10 minutes** :alarm_clock: to go through this list and write an interesting `$updateMany()` query. 
+
+
+:arrow_right: Note that `updateMany()` will make the same modification to all the documents that match the given filter. 
+
+---
+
+## Delete Operations
+
+The MongoDB query language supports 2 delete operations:
+* `deleteOne()`
+* `deleteMany()`
+
+**Exercise 17** :computer: 
 
-Present your results from **Lab 2** and answer any questions your classmates may have.
-Be sure to include your interesting movie :clapper: queries :+1:
+In our `movieDetails` collection, there is only 1 movie from the `year` 2018 with a `title` "QQ Speed". To perform the exercise, delete this record. 
 
+**query:**
+```javascript
+db.movieDetails.deleteOne({"title": "QQ Speed"})
+```
+Alternatively, you could use:
+```javascript
+db.movieDetails.deleteOne({"year": 2018})
+```
+since this filter would also uniquely identify the movie "QQ Speed", considering this collection has only 1 movie from that year. 
+
+**result:**
+```javascript
+{ "acknowledged" : true, "deletedCount" : 1 }
+```
+
+The result confirms that the `deleteOne()` operation was successful, at the same time telling us the number of documents that were deleted. 
+
+**Exercise 18** :computer: 
+
+Our `movieDetails` collection has a few movies from the 1800s!!! Spend **5 minutes** :alarm_clock: to delete all those movies using `deleteMany()`.
+
+---
+
+## Group Activities
```

Comment:
why are we scrapping using other datasets that are already built in favor of them building a poor dataset with only three entries, one which they can't use later?

This is a good exercise for creating documents, and I'm all for it, but it should be meaningful and not arbitrary.  If you want to run an Activity like this, let's given them table data (`.csv` file) and have them convert it to JSON, and THEN run their inserts and such.

Reply:
Sure, that sounds good. You have a few `.csv` files across your repos and since you're more familiar with the data, which ones would you like to include in their group activity:

- [ ] [League of Legends - games.csv](https://github.com/mattdavis0351/mongodb/blob/master/datasets/league_of_legends/games.csv)
- [ ] [candy.csv](https://github.com/mattdavis0351/data/blob/master/candy.csv)
- [ ] [gender.csv](https://github.com/mattdavis0351/data/blob/master/gender.csv) 

I know that `mega city.csv` is being used for their Apache Pig training so I didn't include it in this list!

Reply:
None, those all suck if I remember right.  I think at one point they were used with Talend, but not Mongo.  Nothing stops us from finding some interesting ones though, but these are a no-go

Reply:
we're rolling with what we have!  You already know though